docs/apis/subsystems/roles.md

```diff
@@ -0,0 +1,612 @@
+---
+title: Roles API
+tags:
+  - Roles
+  - Interfaces
+  - Capabilities
+---
+## Definitions
+
+- A **role** is an identifier of the user's status in some context. For example: Teacher, Student and Forum moderator are examples of roles.
+
+- A **capability** is a description of some particular Moodle feature. Capabilities are associated with roles. For example, `mod/forum:replypost` is a capability.
+
+- A **permission** is some value that is assigned for a capability for a particular role.  For example, allow or prevent.
+
+- A **context** is a "space" in the Moodle, such as courses, activity modules, blocks etc.
+
+## The system prior to v1.7
+
+In versions prior to v1.7, Moodle uses a fixed set of roles i.e. primary admin, admins, course creators, editing teachers, non-editing teachers, students, and guests. For each role, the capability or actions that they can perform are fixed. For example, the role student allows the user to submit an assignment, but doesn't allow the user to browse/edit other users' work. By using this setup we limit ourselves to a rather rigid set of capabilities for each role. If we want, say a particular student or group to be able to mark assignments in a particular course, we can't do that without giving these users teacher privileges.
+
+## The new roles and capability system
+
+With v1.7 and greater, Moodle introduces a roles and capabilities system.
+
```

Comment:
Remove this. Roles haven't changed since 2.0. 1.7 is ancient - the final version came out in January 2009, so 13 years since support _ended, and 16 since it stareted

```suggestion
```